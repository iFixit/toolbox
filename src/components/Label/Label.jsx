import React from 'react';
import { string } from 'prop-types';
import styled from 'styled-components';
import constants from '../../constants';

const { spacing, fontSize, lineHeight, color, borderRadius } = constants;

const LabelContainer = styled.span(props => ({
   display: 'inline-block',
   padding: `${spacing[0]} ${spacing[1]}`,
   fontSize: fontSize[0],
   fontWeight: 700,
   lineHeight: lineHeight.title,
   textTransform: 'uppercase',
   color: props.color,
   backgroundColor: props.background,
   borderRadius,
   boxShadow: `inset 0 0 0 1px ${color.grayAlpha[3]}`,
}));

const Label = props => <LabelContainer {...props} />;

Label.propTypes = {
   /** Set the label background color. */
   background: string,
   /** Set the label text color. */
   color: string,
};

Label.defaultProps = {
   background: color.blue[4],
   color: color.white,
};

export default Label;
