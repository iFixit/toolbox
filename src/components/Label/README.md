### Examples

#### Simple Label

```
<Label>Featured</Label>
```

#### Alternate colors

The `<Label>` component exposes `color` and `background` props allowing you to customize the component to your specific use case. Remember to use the colors defined in the `constants` object instead of hard-coding HEX or RGBA values.

```
// NOTE: These colors are hard-coded for demo purposes only. Don't hard-code HEX values in production code.

<Label color="#C27E00" background="#FEF5D9">Pending</Label>
```
