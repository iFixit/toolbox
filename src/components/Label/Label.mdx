---
name: Label
route: /components/label
menu: Components
---

import Label from './Label'
import color from '../../constants/color/color'
import { Playground, PropsTable } from 'docz'

# Label

## Examples

### Default

<Playground>
  <Label>Featured</Label>
</Playground>

### Alternate colors

Remember to use the colors defined in the `constants` object instead of hard-coding HEX or RGBA values.

<Playground>
  <Label color={color.yellow[5]} background={color.yellow[1]}>Pending</Label>
</Playground>

## Props

<PropsTable of={Label} />
