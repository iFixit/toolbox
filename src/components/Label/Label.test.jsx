import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

test('renders without crashing', () => {
   const tree = renderer.create(<Label>Test</Label>).toJSON();
   expect(tree).toMatchSnapshot();
});

test('renders with custom text color and background color', () => {
   const tree = renderer
      .create(
         <Label color="#C27E00" background="#FEF5D9">
            Test
         </Label>,
      )
      .toJSON();
   expect(tree).toMatchSnapshot();
});
