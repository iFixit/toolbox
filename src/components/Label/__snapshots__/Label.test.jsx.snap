// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with custom text color and background color 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1.25;
  text-transform: uppercase;
  color: #C27E00;
  background-color: #FEF5D9;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgba(0, 3, 6, 0.12);
}

<span
  className="glamor-0"
>
  Test
</span>
`;

exports[`renders without crashing 1`] = `
.glamor-0,
[data-glamor-0] {
  display: inline-block;
  padding: 0.25rem 0.5rem;
  font-size: 0.75rem;
  font-weight: 700;
  line-height: 1.25;
  text-transform: uppercase;
  color: #FFFFFF;
  background-color: #0071CE;
  border-radius: 4px;
  box-shadow: inset 0 0 0 1px rgba(0, 3, 6, 0.12);
}

<span
  className="glamor-0"
>
  Test
</span>
`;
